requests/structures.py

```diff
@@ -86,6 +86,128 @@ def __repr__(self):
         return str(dict(self.items()))
 
 
+class HTTPHeaderDict(CaseInsensitiveDict):
```

Comment:
I would like to suggest making use of urllib3's `HTTPHeaderDict` instead o requests having a similar but different version.